### DISCUSSION TITLE
What's the best way to continue to retry RPCs?

### DISCUSSION THREAD

**Discussion by zkSoju** [wevm/viem#272] [General]
I'm getting an error `Details: missing trie node` which seems to be related to the RPC rate limiting.

I'm currently using the default delay and have two RPCs as fallback, but it seems to still fail early (lots of calls).

**Comment by jxom** [comment]
Do your other RPCs in the `fallback` also return `"Details: missing trie node"`? If an RPC in the fallback transport is erroneous, it should fall back to the next one (if the error is indeterministic). However, it this is not the case, we might need to [mark the error as indeterministic](https://github.com/wagmi-dev/viem/blob/main/src/clients/transports/fallback.ts#L87).

**Reply by zkSoju** [reply]
Good catch, it looks like my primary RPC (Alchemy) is falling back to the public one (https://goerli-rollup.arbitrum.io/rpc) and it's the public one that fails. When I remove the public RPC, it works after awhile. Alchemy seems to work fine, just reaches rate limits causing it to fallback I'm guessing.
